echopype/echodata/sensor_ep_version_mapping/v05x_to_v06x.py

```diff
@@ -0,0 +1,1131 @@
+import numpy as np
+import xarray as xr
+
+# TODO: turn this into an absolute import!
+from ...core import SONAR_MODELS
+from ..convention import sonarnetcdf_1
+
+_varattrs = sonarnetcdf_1.yaml_dict["variable_and_varattributes"]
+
+
+def _range_bin_to_range_sample(ed_obj):
+    """
+    Renames the coordinate range_bin to range_sample.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    for grp_path in ed_obj.group_paths:
+
+        if "range_bin" in list(ed_obj[grp_path].coords):
+
+            # renames range_bin in the dataset
+            ed_obj[grp_path] = ed_obj[grp_path].rename(name_dict={"range_bin": "range_sample"})
+
+            ed_obj[grp_path].range_sample.attrs["long_name"] = "Along-range sample number, base 0"
+
+
+def _add_attrs_to_freq(ed_obj):
+    """
+    Makes the attributes of the ``frequency`` variable
+    consistent for all groups. This is necessary because
+    not all groups have the same attributes (some are
+    missing them too) for the ``frequency`` variable.
+    This variable is used to set the variable
+    ``frequency_nominal`` later on.
+
+    Parameters
+    ----------
+    ed_obj : EchoData
+        EchoData object that was created using echopype version 0.5.x
+
+    Notes
+    -----
+    The function directly modifies the input EchoData object.
+    """
+
+    for grp_path in ed_obj.group_paths:
+
+        if "frequency" in list(ed_obj[grp_path].coords):
+
+            # creates consistent frequency attributes
+            # NOTE: the attribute standard_name will be removed in `frequency_nominal`
+            ed_obj[grp_path]["frequency"] = ed_obj[grp_path].frequency.assign_attrs(
+                {
+                    "long_name": "Transducer frequency",
+                    "standard_name": "sound_frequency",
+                    "units": "Hz",
+                    "valid_min": 0.0,
+                }
+            )
+
+
+def _reorganize_beam_groups(ed_obj):
+    """
+    Maps Beam --> Sonar/Beam_group1 and Beam_power --> Sonar/Beam_group2.
```

Comment:
I'm not 100% sure about my recollection (sigh), but I believe an EK80 can have power data only. In that case, I think the only beam group that would exist is Beam_power, and it would map to Sonar/Beam_group1 rather than Sonar/Beam_group2. @leewujung can you confirm?

Reply:
@emiliom I think you are referring to a case I highlighted in your previous [PR](https://github.com/OSOceanAcoustics/echopype/pull/658#discussion_r867061536). If this was what you are indeed referring to, it is not a problem. In v0.5.6 we set this value to `ed.beam`, so this case would get mapped to `Sonar/Beam_group1` (this is correct, it should not be `Sonar/Beam_group2`). This is what should happen we accounted for this in your PR by adding "(either complex samples or uncalibrated power samples)" to the description.

Reply:
Great, thank you.